cognite/client/data_classes/_base.py

```diff
@@ -260,10 +260,21 @@ def _set_null(self, name):
         self._update_object[name] = {"setNull": True}
 
     def _add(self, name, value):
-        self._update_object[name] = {"add": value}
+        update_obj = self._update_object.get(name, {})
+        if "set" in update_obj:
+            del update_obj["set"]
+        if isinstance(value, list):
+            update_obj["add"] = update_obj.get("add", []) + value
```

Comment:
This will fail if the user already has done `update_obj.<name>.add(<dict>)`.

Reply:
In what case will a user reasonably do this?

Reply:
A reasonable user 😮 ❤️

I just think that a double add is in many cases a user error. Maybe they've mixed up two names. I would prefer raising an error on double anything. Combining add and remove makes sense as you say though.

Reply:
I think there's a reasonable case for add in a loop or something, but as long as we give an error it should be easy enough for a user to fix. changes as suggested